edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/TwinManager.cs

```diff
@@ -317,14 +317,13 @@ async Task<TwinInfo> GetTwinInfoWhenCloudOfflineAsync(string id, Exception e)
                 t =>
                 {
                     twinInfo = t;
-                    Events.GetTwinFromStoreWhenOffline(id,
-                        twinInfo.Twin.Properties.Desired.Version,
-                        twinInfo.Twin.Properties.Reported.Version,
-                        e);
+                    Events.GetTwinFromStoreWhenOffline(id, twinInfo, e);
                     return Task.CompletedTask;
                 },
                 () => throw new InvalidOperationException($"Error getting twin for device {id}", e));
-            return twinInfo;
+            return twinInfo.Twin != null
+                ? twinInfo
+                : throw new InvalidOperationException($"Error getting twin for device {id}", e);
```

Comment:
I want to make sure this is the proper place to throw.

Reply:
I am not sure if we can do this. If the device is offline and the device reports something, then the TwinInfo.Twin will be null.